azure_sdk/build.sh

```diff
@@ -0,0 +1,103 @@
+fi
+
+# Copy the repository, add version suffix and update submodules
+if ! [ -d "${PREFIX_AZURE_SRC}" ]; then
+	cp -r "${PREFIX_AZURE_RAW}" "$PREFIX_AZURE_BUILD"
+	mv "${PREFIX_AZURE_BUILD}/${AZURE}" "${PREFIX_AZURE_SRC}"
+	(cd "${PREFIX_AZURE_SRC}" && git submodule update "${update_options[@]}")
+fi
+
+# Apply patches
+for patchfile in "${PREFIX_AZURE_PATCHES}"/*.patch; do
+	if [ ! -f "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied" ]; then
+		echo "applying patch: $patchfile"
+		[[ "${LONG_TEST}" != "y" ]] && [[ $(basename "$patchfile") == "05_c-utility_utests.patch" ]] && continue
+		[[ "${TARGET}" != "armv7m7-imxrt106x"* ]] && [[ "$(basename "$patchfile")" == "04_armv7m7-imxrt106x.patch" ]] && continue
+		patch -d "${PREFIX_AZURE_BUILD}" -p0 -i "$patchfile" && touch "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied"
+	fi
+	# There are some cpp sources, which is currently not supported in Phoenix-RTOS
+	[ "${LONG_TEST}" = "y" ] && rm -rf "${PREFIX_AZURE_SRC}/c-utility/testtools/micromock"
+done
+
+# Set the entered connection string in sample's source (if specified)
+if [ "$AZURE_CONNECTION_STRING" ]; then
+	sed -i "s/\"\[device connection string\]\"/\"${AZURE_CONNECTION_STRING}\"/g" \
+	"${PREFIX_AZURE_SRC}/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c" \
+	&& echo "Connection string has been set properly."
+fi
+
+# Set phoenix c compiler and system root
+PHOENIX_COMPILER_CMD=${CC}
+PHOENIX_SYSROOT=$(${CC} --print-sysroot)
+export PHOENIX_COMPILER_CMD
+export PHOENIX_SYSROOT
+
+# Convert ldflags to format recognizable by gcc, for example -q -> -Wl,-q
+LDFLAGS=$(echo " ${LDFLAGS}" | sed "s/\s/,/g" | sed "s/,-/ -Wl,-/g")
+
+# Build (# http and amqp protocols are currently not supported yet)
+# Treat Phoenix-RTOS as Linux and providing toolchain file was the most suitable solution
+(cd "${PREFIX_AZURE_SRC}/build_all/linux" && ./build.sh \
+--toolchain-file "${PREFIX_AZURE}/toolchain-phoenix.cmake" -cl --sysroot="${PHOENIX_SYSROOT}" \
+--no-http --no-amqp)
+
+# Create azure root directory with provided libs, includes and binaries
+mkdir -p "${PREFIX_AZURE_ROOT}" "${PREFIX_AZURE_LIB}" "${PREFIX_AZURE_INC}" "${PREFIX_AZURE_BIN}"
+
+# Copy azure sdk libraries
+(cd "${PREFIX_AZURE_SRC}/cmake/" && \
+echo "${PREFIX_BUILD}/lib" "${PREFIX_AZURE_LIB}" | xargs -n 1 cp "iothub_client/"*.a "deps/umock-c/libumock_c.a" \
+"umqtt/libumqtt.a" "c-utility/libaziotsharedutil.a")
+
+# Copy azure sdk headers
+(cd "${PREFIX_AZURE_SRC}/" && cp -r "iothub_client/inc/." "deps/umock-c/inc/." \
+"deps/azure-macro-utils-c/inc/." "c-utility/inc/." "${PREFIX_AZURE_INC}")
+cp -r "${PREFIX_AZURE_INC}"/* "${PREFIX_BUILD}/include"
+
+# Install iothub client sample and c-utility tests (if built) in file system
+cp "${PREFIX_AZURE_SRC}/cmake/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample" "${PREFIX_AZURE_BIN}"
```

Comment:
- why do we need to copy it to `PREFIX_AZURE_BIN`?
- please copy unstripped /stripped binaries to `PREFIX_PROG` and `PREFIX_PROG_STRIPPED` respectively

Reply:
- I wanted to have all azure binaries, which we want to use in one place.
 - I see that there is only unstripped version so do You mean to 1) place the unstripped one in `PREFIX_PROG` 2) Strip using `"${CROSS}strip"` 3) Copy to `PROG_STRIPPED` 4) Install it in `/bin`. The mentioned steps could be executed  for all installed binaries so I can create some function for that

Reply:
EDIT:

 - Maybe `PREFIX_AZURE_BIN` is not necessary - I will remove it
 - The proposal with function: 
 ```bash
install_binary () { 
	BINARY_NAME="$(basename "$1")"

	cp "$1" "${PREFIX_PROG}"
	"${CROSS}strip" -s "${PREFIX_PROG}/${BINARY_NAME}" -o "$PREFIX_PROG_STRIPPED/${BINARY_NAME}"
	b_install "${PREFIX_PROG_STRIPPED}/${BINARY_NAME}" /bin
}
```

and then 
```bash
install_binary "${PREFIX_AZURE_SRC}/cmake/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample"
```

and 
```bash
install_binary "$dir/$test_executable"
```